pkg/auth/userregistry/identitymapper/provision.go

```diff
@@ -0,0 +1,153 @@
+package identitymapper
+
+import (
+	"errors"
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrs "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kuser "github.com/GoogleCloudPlatform/kubernetes/pkg/auth/user"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/user"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	identityregistry "github.com/openshift/origin/pkg/user/registry/identity"
+	userregistry "github.com/openshift/origin/pkg/user/registry/user"
+)
+
+type UserNameGenerator func(base string, sequence int) (string, error)
+
+var (
+	MaxGenerateAttempts = 100
+	DefaultGenerator    = UserNameGenerator(func(base string, sequence int) (string, error) {
+		switch {
+		case sequence == 0:
+			return base, nil
+		case sequence > MaxGenerateAttempts:
+			return "", errors.New("Max attempts exceeded")
+		default:
+			return fmt.Sprintf("%s%d", base, sequence+1), nil
+		}
+	})
+)
+
+type provisioningIdentityMapper struct {
+	identity    identityregistry.Registry
+	user        userregistry.Registry
+	generator   UserNameGenerator
+	initializer user.Initializer
+}
+
+// NewAlwaysCreateUserIdentityToUserMapper creates a mapping for the given identity if it already exists
+func NewAlwaysCreateUserIdentityToUserMapper(identityRegistry identityregistry.Registry, userRegistry userregistry.Registry) authapi.UserIdentityMapper {
+	return &provisioningIdentityMapper{identityRegistry, userRegistry, DefaultGenerator, user.NewDefaultUserInitStrategy()}
+}
+
+// UserFor returns info about the user for whom identity info have been provided
+func (p *provisioningIdentityMapper) UserFor(info authapi.UserIdentityInfo) (kuser.Info, error) {
+	ctx := kapi.NewContext()
+
+	identityName := identityregistry.IdentityName(info.GetProviderName(), info.GetProviderUserName())
```

Comment:
Convenience method on the UserIdentity interface for the scope qualified name?  Seems like this approach could produce weird dependencies.